server/test/unit/server/db/test_model.py

```diff
@@ -1,19 +1,232 @@
 """
 Tests for the pulp.server.db.model module.
 """
+from tempfile import mkdtemp
+import shutil
+import os
+
+from mock import patch
+
 try:
     import unittest2 as unittest
 except ImportError:
     import unittest
 
-
 from mongoengine import DateTimeField, DictField, Document, IntField, StringField
 
+from pulp.common import error_codes
+from pulp.devel.unit.util import touch
+from pulp.server.exceptions import PulpCodedException
 from pulp.server.db import model
 from pulp.server.db.model.base import CriteriaQuerySet
 from pulp.server.db.model.fields import ISO8601StringField
 
 
+class TestContentUnit(unittest.TestCase):
+    """
+    Test ContentUnit model
+    """
+
+    def setUp(self):
+        self.working_dir = mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.working_dir, ignore_errors=True)
+
+    def test_model_fields(self):
+        self.assertTrue(isinstance(model.ContentUnit.id, StringField))
+        self.assertTrue(model.ContentUnit.id.primary_key)
+
+        self.assertTrue(isinstance(model.ContentUnit.last_updated, IntField))
+        self.assertTrue(model.ContentUnit.last_updated.required)
+
+        self.assertTrue(isinstance(model.ContentUnit.user_metadata, DictField))
+
+        self.assertTrue(isinstance(model.ContentUnit.storage_path, StringField))
+
+        self.assertTrue(isinstance(model.ContentUnit._ns, StringField))
+        self.assertTrue(model.ContentUnit._ns)
+        self.assertTrue(isinstance(model.ContentUnit.unit_type_id, StringField))
+        self.assertTrue(model.ContentUnit.unit_type_id)
+
+    def test_meta_abstract(self):
+        self.assertEquals(model.ContentUnit._meta['abstract'], True)
+
+    def test_post_init_signal_with_unit_key_fields_defined(self):
+        """
+        Test the init signal handler that validates the existence of the
+        unit_key_fields list
+        """
+        class ContentUnitHelper(model.ContentUnit):
+            unit_key_fields = ['id']
+
+        model.ContentUnit.post_init_signal({}, ContentUnitHelper())
+
+    def test_post_init_signal_without_unit_key_fields_defined(self):
+        """
+        Test the init signal handler that raises an exception if the unit_key_fields list
+        has not been defined.
+        """
+        try:
+            model.ContentUnit.post_init_signal({}, {})
+            self.fail("Previous call should have raised a PulpCodedException")
+        except PulpCodedException, raised_error:
+            self.assertEquals(raised_error.error_code, error_codes.PLP0035)
+
+    def test_pre_save_signal(self):
+        """
+        Test the pre_save signal handler
+        """
+        class ContentUnitHelper(model.ContentUnit):
```

Comment:
Since these tests are on the Contentnit itself, can all of these in-test subclasses be removed, and instead instantiate a ContentUnit() and then set attributes on the instantiated object? I'm trying to separate the test code from the subclassing behavior/concern.